src/Plugin/KeyProvider/PrivateFileKeyProvider.php

```diff
@@ -102,7 +124,15 @@ public function submitConfigurationForm(array &$form, FormStateInterface $form_s
    * {@inheritdoc}
    */
   public function getKeyValue(KeyInterface $key) {
-    if (empty($this->settings->get('file_private_path')) || !is_file($this->getFileUri($key)) || !is_readable($this->getFileUri($key))) {
+    try {
+      $this->checkRequirements($key);
+    }
+    catch (KeyProviderRequirementsException $exception) {
+      $context = [
+        '@message' => (string) $exception,
+      ];
+      $context += Error::decodeException($exception);
+      $this->logger->error('Could not retrieve Apigee Edge authentication key value from the private file storage: @message %function (line %line of %file). <pre>@backtrace_string</pre>', $context);
       return NULL;
```

Comment:
Please add a note that maybe returning `null` is not the best possible option but key's design does not allow to throw an exception and place a reference to this D.o issue: https://www.drupal.org/project/key/issues/3038212